_posts/pl/2021-02-01-odpowiedzi-na-pytania-cd-tech-5.md

```diff
@@ -0,0 +1,114 @@
+W momencie, gdy dostarczymy takową implementację, wówczas Tracking Tokeny zaczną trafiać do bazy.
+Nadal jednak można będzie wykonać replay events (np zapiąć jakiś endpoint na taką akcję), o ile nie wyłączymy tego całkowicie w konfiguracji (opisuję to w kolejnym akapicie).
+
+## Czy można wyłączyć domyślne zachowanie Axona w kontekście odtwarzania eventów przy starcie aplikacji?
+Tak, odpowiedziałem na to pytanie podczas webinaru, lecz w tej chwili mogę wrócić już z konkretnym przykładem w kodzie.
+Podczas transmisji wspominałem o tym, że taka możliwość istnieje i można wręcz wskazać w konfiguracji Event Processor (czyli obiekt odpowiadający za przetwarzanie zdarzeń), który ma być ignorowany podczas odtwarzania.
+
+Event Processory domyślnie dostają nazwę stworzoną na podstawie javowego package, w którym się znajdują (zdaje się, że wyjątkiem są tu Sagi).
+Możemy je też jawnie nazywać z użyciem adnotacji **`@ProcessingGroup`**.
+Mając nazwę processora, w konfiguracji można wykluczyć go z akcji odtwarzania stanu, poprzez sprytne wskazanie, aby Tracking Tokeny dla niego tworzyły się na końcu "jego kolejki".
+Wygląda to następująco:
+```java
+@Configuration
+public class AxonConfig {
+
+    @Autowired
+    public void customTrackingConfig(EventProcessingConfigurer configurer) {
+        var trackingProcessorConfig = TrackingEventProcessorConfiguration
+                .forSingleThreadedProcessing()
+                .andInitialTrackingToken(StreamableMessageSource::createHeadToken);
+
+        // This prevents from replaying in MovieSaga
+        configurer.registerTrackingEventProcessor("MovieSagaProcessor",
+                org.axonframework.config.Configuration::eventStore,
+                c -> trackingProcessorConfig);
+
+        // This prevents from replaying MultipleMoviesRefreshedEvent in RefreshEventHandler
+        configurer.registerTrackingEventProcessor("com.kociszewski.moviekeeper.notreplayable",
+                org.axonframework.config.Configuration::eventStore,
+                c -> trackingProcessorConfig);
+    }
+}
+```
+Opisując powyższy kod językiem ludzkim: klasa **MovieSaga** oraz wszystkie klasy obsługujące zdarzenia, które znajdują się w package'u **com.kociszewski.moviekeeper.notreplayable** zostaną pominięte podczas operacji replay events.
+Uwaga! Jeśli wskażecie package za wysoko, to nie zadziała - musi to być nazwa package'u, w którym znajdują się klasy obsługujące zdarzenia (zagnieżdżenia nie będą uwzględniane).
+
+Namiary na projekt podane są na końcu, jest to ten sam projekt, który omawiałem we wcześniejszych [**wpisach**](https://blog.consdata.tech/authors/mkociszewski.html).
+
+## Jak sobie radzicie w Consdacie z organizacją Event Stormingów w czasach pracy zdalnej?
+Domain Driven Design i Event Storming w Consdacie wciąż jest w fazie R&D, ale z pełnym spokojem mogę polecić [**miro**](https://miro.com) - świetne narzędzie z mnóstwem możliwości.
```

Comment:
Chcemy się tutaj chwalić tym, że robimy webinary z rzeczy, które są R&D i nie korzystamy z nich na co dzień? Może wystarczy polecenie Miro?

Reply:
Jak o tym nie wspomnę, to może wybrzmieć że mamy event stormingi w firmie. Tak mi się wydaje, że to uczciwe jest że wprost mówię, że nie mamy. Jest jeszcze drugie pytanie, które zahacza o ten temat. Ale też się nie upieram

Reply:
Też się nie upieram, po prostu jakbym był na jakimś JUG itd. i ktoś opowiadałby o jakimś temacie, po czym mówił że w jego firmie tego nie używają, to by to dziwnie wyglądało.

Reply:
Wywaliłem to zdanie mówiące o r&d